src/apps/ShidurApp/ShidurToran.js

```diff
@@ -30,22 +30,29 @@ class ShidurToran extends Component {
         this.props.setProps({group});
     };
 
-    closePopup = () => {
+    closePopup = ({disable=false}={}) => {
+        if (disable) {
+            this.disableRoom(this.state.group);
+        }
         this.props.setProps({group: null});
     };
 
-    disableRoom = (e, data, i) => {
+    handleDisableRoom = (e, data) => {
         e.preventDefault();
         if (e.type === 'contextmenu') {
-            let {disabled_rooms} = this.props;
-            let group = disabled_rooms.find(r => r.room === data.room);
-            if(group) return;
-            disabled_rooms.push(data);
-            this.props.setProps({disabled_rooms});
-            this.props.gerGroups();
+            this.disableRoom(data);
         }
     };
 
+    disableRoom(data) {
```

Comment:
Why we does not use **disableRoom = (data) =>** sysntax here?

Reply:
fixed